Update "Build blockchain" overview page (#1651)
LiskHQ/lisk-docs: docs/modules/ROOT/pages/build-blockchain/index.adoc

> = Building a sidechain with the Lisk SDK

Author: I disagree, I think "sidechain" is more fitting in the context, because this cateogry will also include guides how to register the app as a sidechain. So imho it should be either "sidechain" or "sidechain application" in the title.

> = Building a sidechain with the Lisk SDK

Reviewer: Yeah in that case, sidechain application sounds good to me.

Mona Bärenfänger <mona@lightcurve.io>
:toc:
:idprefix:
:idseparator: -
:toc: preamble
// URLs
:url_github_guides_module: https://github.com/LiskHQ/lisk-sdk-examples/tree/development/tutorials/hello/

:url_guide_app: build-blockchain/create-blockchain-app.adoc
:url_guide_config: build-blockchain/configuration.adoc
:url_guide_genesisblock: build-blockchain/create-genesis-block.adoc
:url_guide_module: build-blockchain/module/index.adoc
:url_guide_command: build-blockchain/module/command.adoc
:url_guide_plugin: build-blockchain/create-plugin.adoc

The "Build Blockchain" category contains all the guides related to the development of a sidechain client.

== The Hello World sidechain

By the example of the "Hello World" client, it is explained how to build a very simple sidechain client step by step in the following chapters.

> NOTE: Although the example is kept very simple by intention, the Hello World client implements and uses all the most important components of a module, and is therefore a good example to understand the basics of module development.

Author: ```suggestion
NOTE: Although the example is kept very simple by intention, the Hello World client implements all the most important components, and is therefore a good example to understand the basics of the development of a sidechain client.
```

The "Hello World" sidechain allows each account to save a greeting message on-chain.

> 

Reviewer: ```suggestion
It also stores the addresses of senders of the greeting message, in an off-chain database.
```

The following guides each focus on developing a specific part of the Hello application.

. xref:{url_guide_app}[]: How to bootstrap a default sidechain client.
. xref:{url_guide_config}[]: How to create a custom configuration for a sidechain client.
. xref:{url_guide_module}[]: How to create a new module and how to add it to the client.
In the example, the `HelloModule` will contain all on-chain logic related to the Hello message.
.. xref:{url_guide_command}[]: How to create a new command and how to add it to a module.
In the example, the "Create Hello" command saves Hello messages of accounts, if they send a corresponding transaction.
. xref:{url_guide_genesisblock}[]: How to generate a new genesis block for a sidechain.
. xref:{url_guide_plugin}[]: How to create a new plugin and how to add it to the sidechain client.

> In the example, the plugin will return the latest Hello message which was sent to the network, and its' sender address.

Reviewer: ```suggestion
In the example, the plugin will return the addresses of all the senders who sent a Hello Message to the network.
```

Author: Will it only return the adresses or the Hello messages as well?

Reviewer: We can return both as long they are returned in the event, haven't decoded the event's data yet.

Author: ```suggestion
In the example, the plugin will save all sent Hello messages and their corresponding sender address in an off-chain store, and provide endpoints to retrieve the data from the store.
```

.Complete application code
[TIP]
====
View the complete sample code of the complete "Hello World" sidechain application on GitHub in the {url_github_guides_module}[Lisk SDK examples repository^].
====

The following image provides an overview of the "Hello" application:

> image::build-blockchain/hello-client.png["Hello client overview", 400]

Reviewer 2: imho I would reword the one line in the Stores box to read as follows:

"Is of a maximum length of 300 characters".

> image::build-blockchain/hello-client.png["Hello client overview", 400]

Reviewer: - Please update the description of the plugin as per line 37.
- Also, the plugin will have DB, Endpoint, Schemas, and Configurations, and since this is the overview page of the sidechain client, I think these elements should go into the diagram as well.

Author: @TalhaMaliktz could you update the plugin part of the diagram on Lucidchart? I can then update the diagram in the PR, or it can be updated again in the PR that updates the "How to create a plugin" guide

Reviewer: Sure, will update in during my "How to create a plugin" PR
